LinearizableQueryTest when_multipleQueryLimitIsReachedBeforeHeartbeatAcks_then_noNewQueryIsAccepted

- Fails on `Hazelcast-4.maintenance-Windows-OracleJDK8`
- Fails on [Build #198 (Sep 19, 2020 6:48:00 PM)](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-4.maintenance-Windows-OracleJDK8/198/testReport/junit/com.hazelcast.cp.internal.raft.impl/LinearizableQueryTest/when_multipleQueryLimitIsReachedBeforeHeartbeatAcks_then_noNewQueryIsAccepted/)
- Error
```
expected:<1> but was:<2>
```
- Stacktrace
```
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.cp.internal.raft.impl.LinearizableQueryTest.lambda$when_multipleQueryLimitIsReachedBeforeHeartbeatAcks_then_noNewQueryIsAccepted$0(LinearizableQueryTest.java:179)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1458)
	at com.hazelcast.cp.internal.raft.impl.LinearizableQueryTest.when_multipleQueryLimitIsReachedBeforeHeartbeatAcks_then_noNewQueryIsAccepted(LinearizableQueryTest.java:179)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```
- Seems from high hiccups during execution, yet reporting as I could not recall similar before.
```
Hiccups measured while running test 'when_multipleQueryLimitIsReachedBeforeHeartbeatAcks_then_noNewQueryIsAccepted(com.hazelcast.cp.internal.raft.impl.LinearizableQueryTest):'
20:04:05, accumulated pauses: 737 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:10, accumulated pauses: 725 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:15, accumulated pauses: 730 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:20, accumulated pauses: 728 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:25, accumulated pauses: 724 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:30, accumulated pauses: 737 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:35, accumulated pauses: 726 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:40, accumulated pauses: 726 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:45, accumulated pauses: 748 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:50, accumulated pauses: 723 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:04:55, accumulated pauses: 729 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:00, accumulated pauses: 724 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:05, accumulated pauses: 728 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:10, accumulated pauses: 735 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:15, accumulated pauses: 738 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:20, accumulated pauses: 722 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:25, accumulated pauses: 734 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:30, accumulated pauses: 739 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:35, accumulated pauses: 738 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:40, accumulated pauses: 722 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:45, accumulated pauses: 735 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:50, accumulated pauses: 731 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:05:55, accumulated pauses: 732 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:06:00, accumulated pauses: 735 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:06:05, accumulated pauses: 728 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:06:10, accumulated pauses: 1554 ms, max pause: 835 ms, pauses over 1000 ms: 0
```
- kindly check.

--- a/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/LinearizableQueryTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cp/internal/raft/impl/LinearizableQueryTest.java
@@ -40,9 +40,11 @@ import static com.hazelcast.cp.internal.raft.QueryPolicy.LINEARIZABLE;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.getCommitIndex;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.getLeaderMember;
 import static com.hazelcast.cp.internal.raft.impl.RaftUtil.getLeaderQueryRound;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
@@ -176,7 +178,7 @@ public class LinearizableQueryTest extends HazelcastTestSupport {
         group.start();
 
         RaftNodeImpl leader = group.waitUntilLeaderElected();
-        assertTrueEventually(() -> assertEquals(1, getCommitIndex(leader)));
+        assertTrueEventually(() -> assertThat(getCommitIndex(leader), greaterThanOrEqualTo(1L)));
 
         RaftNodeImpl[] followers = group.getNodesExcept(leader.getLocalMember());
         group.dropMessagesToMember(leader.getLocalMember(), followers[0].getLocalMember(), AppendRequest.class);

